lib/compat/wordpress-6.8/class-gutenberg-rest-post-counts-controller.php

```diff
@@ -0,0 +1,217 @@
+	 */
+	public function __construct() {
+		$this->namespace = 'wp/v2';
+		$this->rest_base = 'counts';
+	}
+
+	/**
+	 * Registers the routes for post counts.
+	 *
+	 * @since 6.8.0
+	 *
+	 * @see register_rest_route()
+	 */
+	public function register_routes() {
+		register_rest_route(
+			$this->namespace,
+			'/' . $this->rest_base . '/(?P<post_type>[\w-]+)',
+			array(
+				'args'   => array(
+					'post_type' => array(
+						'description' => __( 'An alphanumeric identifier for the post type.' ),
+						'type'        => 'string',
+					),
+				),
+				array(
+					'methods'             => WP_REST_Server::READABLE,
+					'callback'            => array( $this, 'get_item' ),
+					'permission_callback' => array( $this, 'get_item_permissions_check' ),
+					'args'                => array(
+						'context' => $this->get_context_param( array( 'default' => 'view' ) ),
+					),
+				),
+				'schema' => array( $this, 'get_public_item_schema' ),
+			)
+		);
+	}
+
+	/**
+	 * Checks if a given request has access to read post counts.
+	 *
+	 * @since 6.8.0
+	 *
+	 * @param WP_REST_Request $request Full details about the request.
+	 * @return true|WP_Error True if the request has read access, WP_Error object otherwise.
+	 */
+	public function get_item_permissions_check( $request ) {
+		$post_type = get_post_type_object( $request['post_type'] );
+
+		if ( ! $post_type ) {
+			return new WP_Error(
+				'rest_invalid_post_type',
+				__( 'Invalid post type.' ),
+				array( 'status' => 404 )
+			);
+		}
+
+		if ( ! current_user_can( $post_type->cap->edit_posts ) ) {
+			return new WP_Error(
+				'rest_cannot_read',
+				__( 'Sorry, you are not allowed to read post counts for this post type.' ),
+				array( 'status' => rest_authorization_required_code() )
+			);
+		}
+
+		return true;
+	}
+
+	/**
+	 * Retrieves post counts for a specific post type.
+	 *
+	 * @since 6.8.0
+	 *
+	 * @param WP_REST_Request $request Full details about the request.
+	 * @return WP_REST_Response|WP_Error Response object on success, or WP_Error object on failure.
+	 */
+	public function get_item( $request ) {
+		$post_type = $request['post_type'];
+		$counts    = wp_count_posts( $post_type );
+
+		if ( ! $counts ) {
+			return new WP_Error(
+				'rest_post_counts_error',
+				__( 'Could not retrieve post counts.' ),
+				array( 'status' => 500 )
+			);
+		}
+
+		$data = $this->prepare_item_for_response( $counts, $request );
+		return rest_ensure_response( $data );
+	}
+
+	/**
+	 * Prepares post counts for response.
+	 *
+	 * @since 6.8.0
+	 *
+	 * @param object          $item    Post counts data.
+	 * @param WP_REST_Request $request Request object.
+	 * @return WP_REST_Response Response object.
+	 */
+	public function prepare_item_for_response( $item, $request ) {
+		$data   = array();
+		$fields = $this->get_fields_for_response( $request );
+
+		foreach ( $fields as $field ) {
+			if ( property_exists( $item, $field ) ) {
+				$data[ $field ] = intval( $item->$field );
```

Comment:
`(int)` type casting is faster than intval().
```suggestion
				$data[ $field ] = (int) $item->$field;
```